### PR TITLE
[layouts] improve right-click popup menu

### PR DESCRIPTION
**Pull request by nirvn**
## Description
@nyalldawson , this PR improves the layout designer window popup by:
- as with the main window, showing bold section (i.e. toolbars, panels) titles
- appending "toolbar" or "panel" to toolbar and panel items (so people don't have to choose between "layout" or "layout" 😉 )
- sorting toolbars and panels alphabetically

Before (left) vs. PR (right):
screenshot from 2018-02-19 15-51-39

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/app/layout/qgslayoutdesignerdialog.cpp [hunk 1]
```diff
@@ -739,6 +740,66 @@ QgsAppLayoutDesignerInterface *QgsLayoutDesignerDialog::iface()
...
+      {
+        panels.append( al[ i ] );
+      }
+      else
+      {
+        toolbars.append( al[ i ] );
+      }
+    }
+
+    std::sort( panels.begin(), panels.end(), cmpByText_ );
```
Comment (nyalldawson) [inline src/app/layout/qgslayoutdesignerdialog.cpp]: Could you make cmpByText_ an lamdba here?

**Comment by nyalldawson** [comment]
Looks good!

**Comment by nirvn** [comment]
@nyalldawson , I've just noticed a few toolbar items were missing from the View -> Toolbars sub-menu, fixed that.

File: src/app/layout/qgslayoutdesignerdialog.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/app/layout/qgslayoutdesignerdialog.cpp]: Yep, fair enough. If it was just in this function I'd rather it was scoped in here (e.g. 

    auto cmpByText_ = [] {... } 

But that's not an option if it's used elsewhere

**Comment by nyalldawson** [comment]
Thanks!

**Comment by nirvn** [comment]
@nyalldawson , this PR made me realize the right-click popup menu was showing report for non-report layouts, and atlas for report layouts. Fixed that too.

**Comment by nirvn** [comment]
@nyalldawson , travis green. All good?

**Comment by nyalldawson** [comment]
Go ahead!

**Pull request merged**